### PR TITLE
Bugfix/issue365 select box

### PR DESCRIPTION
**Pull request by Nova-2119**
### Description of the Change

Altered the logic for deciding whether a vertex is in a selection box so that a vertex is selected if the selection box overlaps any part of it.

### Why Should This Be In Core?

Behaviour of select in box is more intuitive.

### Benefits

Easier to select nodes using a selection box, application is more intuitive to use.

### Possible Drawbacks

Change in behaviour may cause initial confusion

### Applicable Issues

#365

### CODE REVIEW

File: CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/plugins/select/BoxSelectionPlugin.java [hunk 1]
```diff
@@ -146,16 +144,23 @@ public void edit(final GraphWriteMethods graph, final PluginInteraction interact
             }
 
             // Convert world coordinates to camera coordinates.
-            worldLocation.set(x, y, z);
-            worldLocation.subtract(centre);
-            sceneLocation.rotate(worldLocation, rotationMatrix);
-            sceneLocation.setZ(sceneLocation.getZ() - cameraDistance);
+            Vector3f sceneLocation = convertWorldToScene(x, y, z, centre, rotationMatrix, cameraDistance);
+            int rAttr = VisualConcept.VertexAttribute.NODE_RADIUS.get(graph);
+            float r = graph.getFloatValue(rAttr, vxId);
```
Comment (antares1470) [inline CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/plugins/select/BoxSelectionPlugin.java:149]: All 3 can be final

File: CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/plugins/select/BoxSelectionPlugin.java [hunk 2]
```diff
@@ -146,16 +144,23 @@ public void edit(final GraphWriteMethods graph, final PluginInteraction interact
...
-                if (horizontalOffset > left && horizontalOffset < right && verticalOffset < top && verticalOffset > bottom) {
+                final float leftMostPoint = (sceneLocation.getX()-r) / -sceneLocation.getZ();
+                final float rightMostPoint = (sceneLocation.getX()+r) / -sceneLocation.getZ();
+                final float bottomMostPoint = (sceneLocation.getY()-r) / -sceneLocation.getZ();
+                final float topMostPoint = (sceneLocation.getY()+r) / -sceneLocation.getZ();
+                
+                boolean vertexLeftOfBox = rightMostPoint < left;
+                boolean vertexRightOfBox = right < leftMostPoint;
+                boolean vertexBelowBox = topMostPoint < bottom;
+                boolean vertexAboveBox = top < bottomMostPoint;
```
Comment (antares1470) [inline CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/plugins/select/BoxSelectionPlugin.java:161]: All 4 can be final

File: CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/plugins/select/BoxSelectionPlugin.java [hunk 3]
```diff
@@ -146,16 +144,23 @@ public void edit(final GraphWriteMethods graph, final PluginInteraction interact
...
+                
+                if (!(vertexLeftOfBox || vertexRightOfBox || vertexBelowBox || vertexAboveBox)) {
```
Comment (antares1470) [inline CoreInteractiveGraph/src/au/gov/asd/tac/constellation/graph/interaction/plugins/select/BoxSelectionPlugin.java:163]: Are you able to propagate the `!`?
i.e. `!vertexLeftOfBox && !vertexRightOfBox && !vertexBelowBox && !vertexAboveBox`

**Review by antares1470 — CHANGES_REQUESTED** [review]

**Review by arcturus2 — APPROVED** [review]

**Comment by aldebaran30701** [comment]
I've just added those code changes to this PR incase it was waiting on @Nova-2119

**Review by antares1470 — APPROVED** [review]

**Pull request merged**